src/sr_entities_handler.erl

```diff
@@ -70,7 +70,25 @@ resource_exists(Req, State) ->
 -spec content_types_accepted(cowboy_req:req(), state()) ->
   {[{{binary(), binary(), '*'}, atom()}], cowboy_req:req(), state()}.
 content_types_accepted(Req, State) ->
-  {[{{<<"application">>, <<"json">>, '*'}, handle_post}], Req, State}.
+    #{opts := #{path := Path}} = State,
+    {Method, Req2} = cowboy_req:method(Req),
+    try
+        #{metadata := Metadata} = trails:retrieve(Path),
+        AtomMethod = method_to_atom(Method),
+        #{AtomMethod := #{consumes := Consumes}} = Metadata,
+        Handler = compose_handler_name(handle, AtomMethod),
+        Retlist = lists:foldl(fun(Elem, Accin) ->
+                                      [First, Second] =
+                                          string:tokens(Elem, "/"),
```

Comment:
Swagger allows you to specify full [mime types](http://swagger.io/specification/#mimeTypes), like `text/plain; charset=utf-8`. We should use that instead of assuming `*` as the last part of it.
I wonder if cowboy doesn't have a function to parse those same things already.